src/MeshlessFV/MeshlessFVTree.cpp

```diff
@@ -802,24 +783,21 @@ void MeshlessFVTree<ndim,ParticleType,TreeCell>::UpdateGodunovFluxes
 
     // Add all buffers back to main arrays
     // Could be simply atomic?
-#pragma omp barrier
 #pragma omp critical
     {
       for (i=0; i<Nhydro; i++) {
-	    if (mfvdata[i].flags.check_flag(active))
-	      for (k=0; k<ndim+2; k++) mfvdata[i].dQdt[k] += fluxBuffer[i][k];
-        for (k=0; k<ndim+2; k++) mfvdata[i].dQ[k] += dQBuffer[i][k];
-        for (k=0; k<ndim; k++) mfvdata[i].rdmdt[k] += rdmdtBuffer[i][k];
+        if (mfvdata[i].flags.check_flag(active)) {
```

Comment:
So here rdmdt and dQdt are only stored at the beginning/end of the timestep (i.e. when active) whereas dQ can receive contributions throughout the step, whether active or not.  I think this is correct now, but maybe Richard should give a comment before deciding that this is 100% fine?